x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/AbstractLookupService.java

```diff
@@ -278,12 +282,11 @@ private void doLookup(T request, CancellableTask task, ActionListener<List<Page>
         final List<Releasable> releasables = new ArrayList<>(6);
         boolean started = false;
         try {
-
-            ProjectMetadata projMeta = clusterService.state().metadata().getProject();
+            var projectState = projectResolver.getProjectState(clusterService.state());
             AliasFilter aliasFilter = indicesService.buildAliasFilter(
-                clusterService.state().projectState(),
+                projectState,
                 request.shardId.getIndex().getName(),
-                indexNameExpressionResolver.resolveExpressions(projMeta, request.indexPattern)
+                indexNameExpressionResolver.resolveExpressions(projectState.metadata(), request.indexPattern)
```

Comment:
It looks like the main changes are in this file `clusterState.projectState()` -> `projectResolver.getProjectState(clusterState)` and the rest is required to inject `ProjectResolver`.

From what I can see we should avoid:
* org.elasticsearch.cluster.ClusterState#projectState()
* org.elasticsearch.cluster.metadata.Metadata#getProject()

in new code. Anything else we should be aware?

Reply:
Those are definitely the main ones indeed. There are a bunch of methods on `Metadata.Bulider` that are also deprecated and should be removed at some point, see for example: https://github.com/elastic/elasticsearch/blob/876c456ac1a390dd50e5a2a82f95b8a2b862e80f/server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java#L1432-L1436
and there are some other places, but most should be annotated as deprecated.